e2etest/pkg/container/routes.go

```diff
@@ -173,7 +173,14 @@ func BGPRoutes(exc executor.Executor, dev string) (map[netip.Prefix]map[netip.Ad
 		for _, r := range routes {
 			dst, err := netip.ParsePrefix(r.Dst)
 			if err != nil {
-				return nil, fmt.Errorf("invalid prefix %s: %w", r.Dst, err)
+				// `ip route` command does not print the prefix when /32 or /128
```

Comment:
this is definetely too nested :P
Let's move to a parseIPRouteIP function where we can escape early.
Also, the fail and retry approach with the two length is not great, let's check if an IP is v6 or v4 and behave accordingly (or error)